Analog-Devices-MSDK/msdk
# Ble yml rework

This PR improves filtering process of watched files by splitting `BLE` examples and `other` watched files.
When **only** `BLE` examples are modified only those examples are test. 
Anyone time a file in the `others` category is modified then a full device test is warranted. 

Furthermore we only lock boards as needed and unlock in between device tests. 

Lastly when changes to Dats , Datc or Otac, Otas are made it will additionally initiate the connected tests.

We must make sure that after any tests we do the devices are left in a `wiped` state by running mass erase with openocd/gdb
### How to test 
You can test this PR by making a commit to it with changes in any combination of examples across the devices.
You will observe a lot more steps now, the `Check MAX32xxx` step will tell you what tests it found are warranted if any.
The following steps will lock one or two boards if connected tests are needed and then run the tests.
